flink-dstl/flink-dstl-dfs/src/main/java/org/apache/flink/changelog/fs/FsStateChangelogStorage.java

```diff
@@ -63,6 +66,9 @@ public class FsStateChangelogStorage extends FsStateChangelogStorageForRecovery
     private final AtomicInteger logIdGenerator = new AtomicInteger(0);
 
     private final TaskChangelogRegistry changelogRegistry;
+
+    @Nullable private LocalChangelogRegistry localChangelogRegistry = null;
+
     /** The configuration for local recovery. */
     @Nonnull private final LocalRecoveryConfig localRecoveryConfig;
```

Comment:
nit: I'd replace this pair `Nullable registry` and `Nonnull config` with some no-op `LocalChangelogRegistry`.
In `FsStateChangelogWriter.confirm()`, we could just check `if (result.localStreamHandle != null) { ... } `